cetmix_tower_server/models/cx_tower_key_value.py

```diff
@@ -106,10 +104,29 @@ def _read(self, fields):  # pylint: disable=missing-return # doesn't return anyt
                     # (e.g. for missing record or access right)
                     pass
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        """Create a new key value and set secret value"""
+
+        secrets = [v.pop("secret_value", None) for v in vals_list]
+        records = super().create(vals_list)
+        for rec, secret in zip(records, secrets):  # noqa: B905 we need to run on Python 3.10
+            rec._set_secret_value(secret)
+        return records
+
     def write(self, vals):
+        """Write key values"""
+        # Remove secret value from vals and set it later
+        if "secret_value" in vals and not self.env.context.get("write_secret_value"):
+            secret_value = vals.pop("secret_value", None)
+            has_secret_value = True
+        else:
+            has_secret_value = False
         res = super().write(vals)
-        if "secret_value" in vals:
-            self.invalidate_recordset(["secret_value"])
+        # Set secret value
+        if has_secret_value:
+            for key_value in self:
+                key_value._set_secret_value(secret_value)
```

Comment:
let's make the method supported by multi record (check my previous comment for `def write` method)

Reply:
We don't need this currently. Let's leave it as is for now.